### PR TITLE
bs/seq bucketing for prompt and decode

### PR DESCRIPTION
**Pull request by madamczykhabana**
Highlights:
* bs/seq bucketing for both prompt and decode
* warmup
* simplified prompt attention without FSDPA
* common code path between warmup/capture and regular execution
* HPU graph code has been removed and will be reintroduced in next PR

### CODE REVIEW

File: vllm/worker/habana_model_runner.py [hunk 1]
```diff
@@ -778,520 +734,60 @@ def execute_model(
...
+            sampling_params=sampling_params,
+            block_tables=block_tables,
+        )
 
-        # Run the model with the dummy inputs.
-        num_layers = self.model_config.get_num_layers(self.parallel_config)
-        kv_caches = [None] * num_layers
-        self.execute_model(seqs, kv_caches)
-        torch.hpu.synchronize()
+    def profile_run(self) -> None:
```
Comment (kzawora-intel) [inline vllm/worker/habana_model_runner.py:764]: We're risking OOM here - worker uses profile_run to approximate memory model's fwd pass memory usage and uses it as a baseline for KV cache allocation (see habana_worker.py at line 107, determine_num_available_blocks). Without it, we'll allocate more cache blocks and risk crashing with OOM on model execution.

Comment (madamczykhabana) [reply vllm/worker/habana_model_runner.py:764]: Sure, I'll fix it.

File: vllm/worker/habana_model_runner.py [hunk 2]
```diff
@@ -778,520 +734,60 @@ def execute_model(
             logits=logits,
             sampling_metadata=sampling_metadata,
         )
-
+        output.outputs = output.outputs[:real_batch_size]
+        htorch.core.mark_step()
```
[the remark this answers is not in the document]
Comment (madamczykhabana) [reply vllm/worker/habana_model_runner.py:738]: To ensure that model.sample() doesn't get glued together in a single graph with other operations that might follow.
